renderlib/FileReader.cpp

```diff
@@ -52,51 +56,79 @@ FileReader::loadFileDimensions(const std::string& filepath, uint32_t scene)
 {
   std::string extstr = getExtension(filepath);
 
-  if (extstr == ".tif" || extstr == ".tiff") {
+  if (filepath.find("http") == 0) {

```

Comment:
These if cascades worry me, but are probably good enough for an initial version.

Reply:
What is the concern here? Is it the repetition?
There's a future refactor where maybe the file type is determined much earlier and in one place only (in maybe a factory method) and then a derived-class reader implementation is selected that way.

Reply:
Both the repetition and the cascading if/else set that is hardcoded.  I'd prefer a lookup table somewhere independent of the rest of the stuff.  But maybe that is over-complicating things; the if/else cascade is very clear, at least.